src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs

```diff
@@ -91,10 +91,19 @@ void RecoverPendingTransactions()
             }
             else
             {
-                if (Directory.Exists(pendingTransactionDir))
+                if (!Directory.Exists(pendingTransactionDir))
+                {
+                    return;
+                }
+
+                try
                 {
                     Directory.Delete(pendingTransactionDir, true);
                 }
+                catch (IOException e)
```

Comment:
Should this also be changed to `Exception`?

Reply:
Good catch.